src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrenceEvaluator.cs

```diff
@@ -0,0 +1,1115 @@
+        {
+            RecurrencePattern pattern = settings.Recurrence.Pattern;
+
+            DateTimeOffset start = settings.Start.Value;
+
+            Debug.Assert(time >= start);
+
+            int interval = pattern.Interval;
+
+            DateTime alignedTime = time.DateTime + start.Offset - time.Offset;
+
+            int yearGap = alignedTime.Year - start.Year;
+
+            if (alignedTime.TimeOfDay + TimeSpan.FromDays(alignedTime.DayOfYear) < start.TimeOfDay + TimeSpan.FromDays(start.DayOfYear))
+            {
+                //
+                // E.g. start: 2023-9-1T12:00:00 and time: 2024-9-1T11:00:00
+                // Not a complete year
+                yearGap -= 1;
+            }
+
+            int numberOfInterval = yearGap / interval;
+
+            previousOccurrence = start.AddYears(numberOfInterval * interval);
+
+            numberOfOccurrences = numberOfInterval + 1;
+        }
+
+        /// <summary>
+        /// Find the closest previous recurrence occurrence before the provided timestamp according to the "RelativeYearly" recurrence pattern.
+        /// <param name="time">A timestamp.</param>
+        /// <param name="settings">The settings of time window filter.</param>
+        /// <param name="previousOccurrence">The closest previous occurrence.</param>
+        /// <param name="numberOfOccurrences">The number of occurrences between the time and the recurrence start.</param>
+        /// </summary>
+        private static void FindRelativeYearlyPreviousOccurrence(DateTimeOffset time, TimeWindowFilterSettings settings, out DateTimeOffset previousOccurrence, out int numberOfOccurrences)
+        {
+            RecurrencePattern pattern = settings.Recurrence.Pattern;
+
+            DateTimeOffset start = settings.Start.Value;
+
+            Debug.Assert(time >= start);
+
+            int interval = pattern.Interval;
+
+            DateTime alignedTime = time.DateTime + start.Offset - time.Offset;
+
+            int yearGap = alignedTime.Year - start.Year;
+
+            if (alignedTime.Month < start.Month)
+            {
+                //
+                // E.g. start: 2023.9 and time: 2024.8
+                // Not a complete year
+                yearGap -= 1;
+            }
+            else if (alignedTime.Month == start.Month && 
+                !pattern.DaysOfWeek.Any(day =>
+                    alignedTime >= NthDayOfWeekInTheMonth(alignedTime, pattern.Index, day) + start.TimeOfDay))
+            {
+                //
+                // E.g. start: 2023.9.1 (the first Friday in 2023.9) and time: 2024.9.2 (the first Friday in 2023.9 is 2024.9.6)
+                // Not a complete year
+                yearGap -= 1;
+            }
+
+            int numberOfInterval = yearGap / interval;
+
+            DateTime previousOccurrenceMonth = start.AddYears(numberOfInterval * interval).DateTime;
```

Comment:
```suggestion
            DateTime previousOccurrenceMonth = start.AddYears(yearGap).DateTime;
```

Looks like it's the same value so maybe you can just use `yearGap`

Reply:
Actually, numberOfInterval * interval may not equal to yearGap. 

For example, if yearGap is 5 and interval is 2
int numberOfInterval = yearGap / interval; // this will be 2

Reply:
Missed that, my fault